astropy/visualization/scripts/fits2bitmap.py

```diff
@@ -100,14 +100,20 @@ def fits2bitmap(
     if image.ndim != 2:
         log.critical(f"data in FITS extension {ext} is not a 2D array")
 
+    if isinstance(filename, str):
+        filename = Path(filename)
+
     if out_fn is None:
-        out_fn = os.path.splitext(filename)[0]
-        if out_fn.endswith(".fits"):
-            out_fn = os.path.splitext(out_fn)[0]
-        out_fn += ".png"
+        out_fn = filename.parent / filename.stem
+
+        if out_fn.suffix == ".fits":
+            out_fn = out_fn.parent / out_fn.stem
+        out_fn = out_fn.with_suffix(".png")
```

Comment:
If `out_fn` still has a suffix then the new code is not equivalent to the old code.